sdk/storage/AzureStorageBlob/Source/Operation/BlobDownloadOperations.swift

```diff
@@ -100,9 +100,16 @@ internal class BlobDownloadInitialOperation: TransferOperation {
 
         let group = DispatchGroup()
         group.enter()
-        downloader.initialRequest { result, _ in
+        downloader.initialRequest { result, httpResponse in
             switch result {
             case .success:
+                guard let responseHeaders = httpResponse?.headers else {
+                    assertionFailure("Response headers not found.")
+                    return
+                }
+                let blobProperties = BlobProperties(from: responseHeaders)
+                parent.totalBytesToTransfer = Int64(downloader.totalSize)
```

Comment:
Does `downloader.progress` need to be updated here too? It's a little confusing why some of this logic is here & in BlockOperations vs. being handled internally in the downloader itself.

Reply:
No, this stuff gets computed inside the downloader. It has to be copied to the transfer object because the downloader is not persisted. I did update this logic to pull the bytes transferred from the downloader instead of serializing the headers.